src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,207 @@
+                try {
+                    Task newTask = new Todo(arr[1]);
+                    ls.add(newTask);
+                    ui.showTodoMsg(ls, newTask);
+                } catch (Exception ex) {
+                    ui.showTodoError();
+                }
+                break;
+            case DEADLINE:
+                try {
+                    String[] arrOfStr = Parser.parse(arr, 1);
+                    try {
+                        Task newTask = new Deadline(arrOfStr[0], arrOfStr[1]);
+                        ls.add(newTask);
+                        ui.showDeadlineEventMsg(ls, newTask);
+                    } catch (Exception ex) {
+                        ui.showDeadlineFormatError();
+                    }
+                } catch (Exception ex) {
+                    ui.showDeadlineError();
+                }
+                break;
+            case EVENT:
+                try {
+                    String[] arrOfStr = Parser.parse(arr, 2);
+                    try {
+                        Task newTask = new Event(arrOfStr[0], arrOfStr[1]);
+                        ls.add(newTask);
+                        ui.showDeadlineEventMsg(ls, newTask);
+                    } catch (Exception ex) {
+                        ui.showEventFormatError();
+                    }
+                } catch (Exception ex) {
+                    ui.showEventError();
+                }
+                break;
+            case DELETE:
+                try {
+                    int numToBeDeleted = Parser.parseInt(str);
+                    Task tsk = ls.get(numToBeDeleted);
+                    ls.remove(numToBeDeleted);
+                    ui.showDeleteMsg(ls, tsk);
+                } catch (Exception ex) {
+                    ui.showDeleteError();
+                }
+                break;
+            case CHECK:
+                try {
+                    String checkDate = arr[1];
+                    LocalDate date = LocalDate.parse(checkDate);
+                    if (ls.isEmpty()) {
+                        ui.showCheckNoTask();
+                    } else {
+                        ui.horizontalDiv();
+                        int counter = 0;
+                        for (int i = 0; i < ls.size(); i++) {
+                            Task task = ls.get(i);
+                            if (task instanceof Event) {
+                                if (((Event) task).getAt().equals(date) && !task.isDone()) {
+                                    counter += 1;
+                                    System.out.println(counter + ". " + task.toString());
+                                }
+                            } else if (task instanceof Deadline && !task.isDone()) {
+                                if (((Deadline) task).getDate().equals(date)) {
+                                    counter += 1;
+                                    System.out.println(counter + ". " + task.toString());
+                                }
```

Comment:
Probably you could consider how to differentiate between Event, Deadline by overloading a method in your Task class instead of using instanceOf. Using instanceOf is normally a last resort so please consider using an overloaded method to distinguish between a deadline, task and even todo.